tomcat/src/main/java/org/apache/coyote/http11/RequestData.java

```diff
@@ -0,0 +1,105 @@
+package org.apache.coyote.http11;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class RequestData {
+
+    private final HttpMethod httpMethod;
+    private final String resource;
+    private final HttpVersion httpVersion;
+    private final HttpContentType httpContentType;
+    private final Map<String, String> queryParameter;
+
+    private RequestData(HttpMethod httpMethod,
+                        String resource,
+                        HttpVersion httpVersion,
+                        HttpContentType httpContentType,
+                        Map<String, String> queryParameter) {
+        this.httpMethod = httpMethod;
+        this.resource = resource;
+        this.httpVersion = httpVersion;
+        this.httpContentType = httpContentType;
+        this.queryParameter = queryParameter;
+    }
+
+    public static RequestData of(List<String> rawHttpRequest) {
+        String[] firstLine = rawHttpRequest.get(0).split(" ");
+        HttpMethod httpMethod = HttpMethod.getHttpMethod(firstLine[0].trim());
+        String resource = firstLine[1].trim();
+        Map<String, String> queryParameter = new HashMap<>();
+        if (resource.contains("?")) {
+            getQueryParameter(resource, queryParameter);
+            resource = resource.substring(0, resource.indexOf("?"));
+        }
+        HttpVersion httpVersion = HttpVersion.getHttpVersion(firstLine[2].trim());
+        HttpContentType httpContentType = getHttpContentType(rawHttpRequest);
+        return new RequestData(httpMethod, resource, httpVersion, httpContentType, queryParameter);
+    }
+
+    private static void getQueryParameter(String resource, Map<String, String> queryParameter) {
+        String queryString = resource.substring(resource.indexOf("?") + 1);
+        for (String pair : queryString.split("&")) {
+            String[] keyValue = pair.split("=");
+            if (keyValue.length == 2) {
+                queryParameter.put(keyValue[0], keyValue[1]);
+            }
+        }
```

Comment:
GET /index?calc=1+1=2 http/1.1 과 같은 path에서 
String[] keyValue = pair.split("="); 를 사용하면 [calc, 1+1, 2] 가 나옵니다.

다음처럼 최대 스플릿 횟수를 지정해주세요.
```suggestion
        for (String pair : queryString.split("&")) {
            String[] keyValue = pair.split("=", 2);
            if (keyValue.length == 2) {
                queryParameter.put(keyValue[0], keyValue[1]);
            }
        }
```

Reply:
와우 split 최대 갯수를 지정해줄 수 있군요!
대박 감사합니다 반영했습니다.!

Reply:
와우 split 최대 갯수를 지정해줄 수 있군요!
대박 감사합니다 반영했습니다.!